programs/bpf_loader/src/syscalls.rs

```diff
@@ -2118,976 +2285,3475 @@ fn call<'a>(
+    cost: u64,
+    compute_meter: Rc<RefCell<dyn ComputeMeter>>,
+    loader_id: &'a Pubkey,
+}
+impl<'a> SyscallObject<BpfError> for SyscallBigNumFromDecStr<'a> {
+    fn call(
+        &mut self,
+        in_dec_str_addr: u64,
+        in_size: u64,
+        out_array_addr: u64,
+        out_size_addr: u64,
+        out_negative_addr: u64,
+        memory_mapping: &MemoryMapping,
+        result: &mut Result<u64, EbpfError<BpfError>>,
+    ) {
+        question_mark!(
+            self.compute_meter
+                .consume(self.cost + calc_bignum_cost!(in_size as f64)),
+            result
+        );
 
-    macro_rules! assert_access_violation {
-        ($result:expr, $va:expr, $len:expr) => {
-            match $result {
-                Err(EbpfError::AccessViolation(_, _, va, len, _)) if $va == va && len == len => (),
-                _ => panic!(),
-            }
+        // Get the string and convert to BigNum
+        let in_dec_raw = question_mark!(
+            translate_slice::<u8>(
+                memory_mapping,
+                in_dec_str_addr,
+                in_size,
+                self.loader_id,
+                true
+            ),
+            result
+        );
+        let i = match in_dec_raw.iter().position(|byte| *byte == 0) {
+            Some(i) => i,
+            None => in_size as usize,
         };
+        let big_number = BigNum::from_dec_str(from_utf8(&in_dec_raw[..i]).unwrap()).unwrap();
+        let big_number_bytes = big_number.as_ref().to_vec();
+        let big_number_len = big_number_bytes.len() as u64;
+        let out_size = question_mark!(
+            translate_type_mut::<u64>(memory_mapping, out_size_addr, self.loader_id, true),
+            result
+        );
+
+        // Get result
+        let byte_slice = question_mark!(
+            translate_slice_mut::<u8>(
+                memory_mapping,
+                out_array_addr,
+                *out_size,
+                self.loader_id,
+                true
+            ),
+            result
+        );
+        // Get negative flag pointer
+        let is_negative = question_mark!(
+            translate_type_mut::<u64>(memory_mapping, out_negative_addr, self.loader_id, true),
+            result
+        );
+        // Get return bytes
+        // Exceeds allocated size
+        if big_number_len > *out_size {
+            *result = Err(SyscallError::BigNumberToBytesError.into())
+        } else {
+            // Equal (memcpy)
+            if big_number_len == *out_size {
+                (*byte_slice).copy_from_slice(&big_number_bytes);
+            }
+            // Smaller than our buffer
+            else {
+                let mut index = 0;
+                for byte in big_number_bytes.iter() {
+                    (*byte_slice)[index] = *byte;
+                    index += 1;
+                }
+                *out_size = index as u64;
+            }
+            *is_negative = big_number.is_negative() as u64;
```

Comment:
When calling big_number.as_slice() the sign information is lost?  I would think that would be someone encoded in the slice

Reply:
BigNum `to_vec()` docs:
`/// Returns a big-endian byte vector representation of the absolute value of `self`.`

So it's just the absolute number.